sql/src/main/java/io/crate/planner/operators/Insert.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.operators;
+
+import io.crate.analyze.OrderBy;
+import io.crate.analyze.relations.AbstractTableRelation;
+import io.crate.analyze.relations.QueriedRelation;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.planner.PlannerContext;
+import io.crate.planner.Merge;
+import io.crate.planner.ExecutionPlan;
+import io.crate.planner.projection.MergeCountProjection;
+import io.crate.planner.projection.Projection;
+import io.crate.planner.projection.builder.ProjectionBuilder;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class Insert implements LogicalPlan {
+
+    private final LogicalPlan source;
+    private final QueriedRelation relation;
+    private final Projection projection;
+
+    public Insert(LogicalPlan source, QueriedRelation relation, Projection projection) {
```

Comment:
Is it possible to rename the projection here to something more specific? What kind of projection is this?

Would it make sense to build the projection inside the operator as well?